docs/toucan-form/changeset-validation/index.md

```diff
@@ -0,0 +1,48 @@
+---
+title: Changeset validation
+order: 2
+---
+
+# Ember Changeset validation
+
+This demo shows how to implement [ember-changeset](https://github.com/poteto/ember-changeset) validation with ember-toucan-form, powered by [ember-headless-form](https://ember-headless-form.pages.dev/docs/validation/ember-changeset).
+
+## Install the adapter package
+
+Before using ember-changeset validations with Toucan Form, you'll need to install it as a dependency.
+
+```bash
+pnpm add ember-headless-form-changeset
+# or
+yarn add ember-headless-form-changeset
+# or
+npm install ember-headless-form-changeset
+```
+
+## Default initial validation
+
+By default, Toucan Form will set the `@validateOn` argument to `focusout`; however, this can be overriden by providing your own `@validateOn` component argument [supported by ember-headless-form](https://ember-headless-form.pages.dev/docs/validation/timing#validateon).
+
+```hbs
+<ToucanForm @validateOn='input' />
+```
+
+## Form data and TypeScript
+
+The provided form `@data` argument must be properly typed if you are using TypeScript. Follow the ember-headless-form [typing of form data docs](https://ember-headless-form.pages.dev/docs/typescript#typing-of-form-data) for more information. Essentially your `@data` argument type must properly match the form fields used in your templates.
+
+```ts
+interface MyFormData {
+  firstName?: string;
+  lastName?: string;
+}
+
+const data: MyFormData = {};
+```
+
+```hbs
+<ToucanForm @data={{data}} as |form|>
+  <form.Input @label='First name' @name='firstName' />
+  <form.Input @label='Last name' @name='lastName' />
+</ToucanForm>
+```
```

Comment:
You'll notice this section is copied across here and the native-validation docs as well.  I was thinking about putting these in the root https://f37cfe54.ember-toucan-core.pages.dev/docs/toucan-form, but I personally tend to go to the code examples with documentation anyway.  I'd be afraid if we put this information at https://f37cfe54.ember-toucan-core.pages.dev/docs/toucan-form, people just won't read it, as they _maybe_ only care about a) looking at code b) either the changeset or native demos anyway.

Any thoughts?  I don't have strong opinions here, more just wanted to share what I was thinking.